### PR TITLE
CB-12613 Added temporaryStorage field to InstanceTemplate in DataHub post API

### PR DESCRIPTION
**Pull request by daszabo**
This field will controll if only ephemeral storage should be initialized for VMs

### CODE REVIEW

**Review by horadla23 — APPROVED** [review]
LGTM

**Comment by horadla23** [comment]
one minor thing: in entity, please add a `DefaultEnumConverter` to the new field, like here: https://github.com/hortonworks/cloudbreak/blob/89411c9e1676344c29ca813407f41694aed95fcf/core-model/src/main/java/com/sequenceiq/cloudbreak/domain/Template.java#L49

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/Template.java [hunk 1]
```diff
@@ -68,6 +69,8 @@
     @OneToMany(mappedBy = "template", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
     private Set<VolumeTemplate> volumeTemplates;
 
+    private TemporalStorage temporalStorage;
```
Comment (horadla23) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/Template.java:72]: sql is missing for this, isn't it?
can this value be null?

**Review by horadla23 — CHANGES_REQUESTED** [review]
i approved this first, then those questions above came into my mind

File: common/src/main/java/com/sequenceiq/cloudbreak/common/type/TemporalStorage.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.sequenceiq.cloudbreak.common.type;
+
+public enum TemporalStorage {
```
Comment (lajosrodek) [inline common/src/main/java/com/sequenceiq/cloudbreak/common/type/TemporalStorage.java:3]: Did not you rather mean `temporary`? "Temporal" means "time-dependent" or "related to time". 😄

**Review by horadla23 — APPROVED** [review]

File: common/src/main/java/com/sequenceiq/cloudbreak/common/type/TemporalStorage.java (hunk 2 above)
Comment (daszabo) [reply common/src/main/java/com/sequenceiq/cloudbreak/common/type/TemporalStorage.java]: Fair point, thanks :)

**Review by lajosrodek — APPROVED** [review]
LGTM, nice work!

**Pull request merged**